src/goto-programs/goto_contractor.cpp

```diff
@@ -167,12 +219,272 @@ void goto_contractort::apply_contractor()
   log_status("{}", oss.str());
 }
 
-ibex::Ctc *goto_contractort::create_contractor_from_expr2t(const expr2tc &expr)
+bool goto_contractort::initialize_main_function_loops()
+{
+  auto it = goto_functions.function_map.find("c:@F@main");
```

Comment:
What about if we model check C++ programs? We should use something like has_prefix.

Reply:
Not sure i understand this. May i ask for an example?

Reply:
What happens with C++ programs? Do we still use c:@F@main?

Reply:
There is `config.main` holding the value of the `--function` parameter, which is the generic way determine the entry point of the program. However, there are a few users of `c:@F@main`, hence this might be a broader issue: 
![image](https://github.com/esbmc/esbmc/assets/3314468/49214631-592f-4315-8de5-12a3462329ab)